### DISCUSSION TITLE
BT 2.38.0 and Aria automation 8.16.2 custom form issue

### DISCUSSION THREAD

**Discussion by pe1pip** [vmware/build-tools-for-vmware-aria#257] [Q&A]
Hi all,

we're having big problems with vRA 8.16.2 and custom forms for our cloud template based catalog items. It seems that we can't get a 'unversioned' custom form uploaded. BT 2.38.0 report:

```
[ERROR] Failed to execute goal com.vmware.pscoe.maven.plugins:vrealize-package-maven-plugin:2.38.0:push (default-cli) on project virtualMachine: Execution default-cli of goal com.vmware.pscoe.maven.plugins:vrealize-package-maven-plugin:2.38.0:push failed: org.springframework.web.client.HttpClientErrorException: 500 Invalid/Unreachable FQDN or IP address: 500 Internal Server Error
[ERROR] Cache-Control: no-cache, no-store, max-age=0, must-revalidate
[ERROR] Content-Length: 193
[ERROR] Content-Type: application/json
[ERROR] Expires: 0
[ERROR] Pragma: no-cache
[ERROR] Referrer-Policy: no-referrer
[ERROR] Strict-Transport-Security: max-age=31536000 ; includeSubDomains
[ERROR] X-Content-Type-Options: nosniff
[ERROR] X-Frame-Options: SAMEORIGIN
[ERROR] X-Xss-Protection: 1 ; mode=block
[ERROR] Date: Wed, 03 Apr 2024 09:04:35 GMT
[ERROR] 
[ERROR] {"timestamp":"2024-04-03T09:04:35.404+0000","path":"/form-service/api/forms","status":500,"error":"Internal Server Error","requestId":"0520145a-119530","@type":"java.lang.NullPointerException"}
```

Has anyone else seen this? Is this an issue in vRA or is there a fix needed in the build tools to deal with this new minor update to Arai Automation?

**Comment by pe1pip** [comment]
nothing? Nobody?

**Comment by pe1pip** [comment]
vRA is broken when a complex is input to another form action

**Comment by sugeeshC** [comment]
I attempted to reproduce the issue mentioned in this discussion using vRBT 2.38.0 and vRA version 8.16.0.3369, but couldn't reproduce the error as mentioned.

While I did not encounter the original error, but obeserved a new inconsistency in how non-versioned custom forms handled in vRBT and reported [here as a bug](https://github.com/vmware/build-tools-for-vmware-aria/issues/342).

To better understand and investigate the original issue, @pe1pip  could you please provide more details or the exact resources you used ?
